modules/sdk-coin-eth/src/eth.ts

```diff
@@ -1204,6 +1230,369 @@ export class Eth extends BaseCoin {
+    const batchData = optionalDeps.ethUtil.addHexPrefix(
+      this.getMethodCallData(BATCH_METHOD_NAME, BATCH_METHOD_TYPES, batchExecutionInfo.values).toString('hex')
+    );
+
+    // Get sequence ID using contract call
+    // we need to wait between making two polygonscan calls to avoid getting banned
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+    const sequenceId = await this.querySequenceId(walletContractAddress);
+
+    const txInfo = {
+      recipients: recipients,
+      expireTime: this.getDefaultExpireTime(),
+      contractSequenceId: sequenceId,
+      gasLimit: gasLimit.toString(10),
+      isEvmBasedCrossChainRecovery: true,
+    };
+
+    const network = this.getNetwork();
+    const batcherContractAddress = network?.batcherContractAddress as string;
+
+    const txBuilder = this.getTransactionBuilder() as TransactionBuilder;
+    txBuilder.counter(bitgoFeeAddressNonce);
+    txBuilder.contract(walletContractAddress);
+    let txFee;
+    if (params.eip1559) {
+      txFee = {
+        eip1559: {
+          maxPriorityFeePerGas: params.eip1559.maxPriorityFeePerGas,
+          maxFeePerGas: params.eip1559.maxFeePerGas,
+        },
+      };
+    } else {
+      txFee = { fee: gasPrice.toString() };
+    }
+    txBuilder.fee({
+      ...txFee,
+      gasLimit: gasLimit.toString(),
+    });
+
+    const transferBuilder = txBuilder.transfer() as TransferBuilder;
+
+    transferBuilder
+      .amount(batchExecutionInfo.totalAmount)
+      .contractSequenceId(sequenceId)
+      .expirationTime(this.getDefaultExpireTime())
+      .to(batcherContractAddress)
+      .data(batchData);
+
+    if (params.walletPassphrase) {
+      txBuilder.transfer().key(userSigningKey);
+    }
+
+    const tx = await txBuilder.build();
+
+    const response: OfflineVaultTxInfo = {
+      txHex: tx.toBroadcastFormat(),
+      userKey,
+      coin: this.getChain(),
+      gasPrice: optionalDeps.ethUtil.bufferToInt(gasPrice).toFixed(),
+      gasLimit,
+      recipients: txInfo.recipients,
+      walletContractAddress: tx.toJson().to,
+      amount: batchExecutionInfo.totalAmount,
+      backupKeyNonce: bitgoFeeAddressNonce,
+      eip1559: params.eip1559,
+    };
+    _.extend(response, txInfo);
+    response.nextContractSequenceId = response.contractSequenceId;
+
+    if (params.walletPassphrase) {
+      const halfSignedTxn: HalfSignedTransaction = {
+        halfSigned: {
+          txHex: tx.toBroadcastFormat(),
+          recipients: txInfo.recipients,
+          expireTime: txInfo.expireTime,
+        },
+      };
+      _.extend(response, halfSignedTxn);
+
+      const feesUsed: FeesUsed = {
+        gasPrice: optionalDeps.ethUtil.bufferToInt(gasPrice).toFixed(),
+        gasLimit: optionalDeps.ethUtil.bufferToInt(gasLimit).toFixed(),
+      };
+      response['feesUsed'] = feesUsed;
+    }
+
+    return response;
+  }
+
+  async recoverEthLikeTokenforEvmBasedRecovery(
```

Comment:
I can see a lot of code duplication which can be eliminated
I'm okay if you can do it in a follow up PR

Reply:
Yeah, I agree. I will do code refactoring in a follow-up PR.